examples/c/pipe-ping-pong.c

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/* This should come first. */
+#define _POSIX_C_SOURCE 200809L
```

Comment:
What does this number mean?

Reply:
@BrianZill,

In Linux, this is required if one wants to use a `struct timespec`. If you remove this, the compiler will not be able to locate the definition of `struct timespec`

This is documented in the Linux manual page [[1](https://man7.org/linux/man-pages/man7/feature_test_macros.7.html)]:

```
       _POSIX_C_SOURCE
              Defining this macro causes header files to expose
              definitions as follows:

              •  The value 1 exposes definitions conforming to
                 POSIX.1-1990 and ISO C (1990).

              •  The value 2 or greater additionally exposes definitions
                 for POSIX.2-1992.

              •  The value 199309L or greater additionally exposes
                 definitions for POSIX.1b (real-time extensions).

              •  The value 199506L or greater additionally exposes
                 definitions for POSIX.1c (threads).

              •  (Since glibc 2.3.3) The value 200112L or greater
                 additionally exposes definitions corresponding to the
                 POSIX.1-2001 base specification (excluding the XSI
                 extension).  This value also causes C95 (since glibc
                 2.12) and C99 (since glibc 2.10) features to be exposed
                 (in other words, the equivalent of defining
                 _ISOC99_SOURCE).

              •  (Since glibc 2.10) The value 200809L or greater
                 additionally exposes definitions corresponding to the
                 POSIX.1-2008 base specification (excluding the XSI
                 extension).
```

Note that these constants are also declared in the existing test files (ie. `tcp-ping-pong`, `tcp-push-pop`, `udp-ping-pong` and `udp-push-pop`).

Reply:
Could you expand the comment preceding this line to explain this?  Something like:

// Glibc macro to expose definitions corresponding to the POSIX.1-2008 base specification.
// See https://man7.org/linux/man-pages/man7/feature_test_macros.7.html.

Or some such?